### DISCUSSION TITLE
Grouping options of a Select field with a relationship

### DISCUSSION THREAD

**Discussion by dev-dev-15** [filamentphp/filament#9471] [Help] [answered]
The Select field already has the grouping options feature:
![image](https://github.com/filamentphp/filament/assets/14108109/a06ff5cf-3160-4ef8-84fa-b1e7fd6fa77d)
But I want to use that feature in combination with a relationship.
Eg: in an Invoice Filament resource creation page, I'll have a Select field with the relationship "prices", and the prices should be grouped by their "products" which is a HasMany relationship (Products has many prices - price belongs to product), and of course the user will select the prices not  the products, just like Stripe:
![image](https://github.com/filamentphp/filament/assets/14108109/9e0093d9-9788-4ecf-9731-7e73a161a06b)

**Comment by sinnbeck** [comment, marked as answer]
If the nested relationship is loaded already you could probably do something like this.

```php
->options(fn($record) => $record->prices->groupBy('product.name')->map->pluck('name', 'id))
#or
->options(fn() => Price::with('products')->get()->groupBy('product.name')->map->pluck('name', 'id))

**Reply by sinnbeck** [reply]
Or perhaps something like this (untested)
```php
->options(fn($record) => $record->products->keyBy('name')->map->pluck('prices.name', 'prices.id))

**Reply by dev-dev-15** [reply]
I'll give it a shot, thanx
